### ISSUE TITLE
`sbt publishSigned` command fails with 1.0.0-RC8

### ISSUE THREAD

**Issue by seratch** [coursier/coursier#619]
I've encountered an error like `java.lang.IllegalStateException: Ivy file not found in cache for {artifact}` when running `sbt publishSigned` while enabling sbt-coursier 1.0.0-RC8. 

The problem doesn't happen with 1.0.0-RC6.

Let me paste the output when the issue occurred with my project: https://github.com/skinny-framework/skinny-framework

```
$ ./publish.sh
[warn] Executing in batch mode.
[warn]   For better performance, hit [ENTER] to switch to interactive mode, or
[warn]   consider launching sbt without any commands, or explicitly passing 'shell'
[info] Loading project definition from /Users/kazuhirosera/github/oss/skinny-framework/project
[info] Resolving key references (14024 settings) ...
[info] Set current project to skinny-framework (in build file:/Users/kazuhirosera/github/oss/skinny-framework/)
[info] Setting version to 2.12.2
[info] Reapplying settings...
[info] Set current project to skinny-framework (in build file:/Users/kazuhirosera/github/oss/skinny-framework/)
[success] Total time: 16 s, completed Jul 22, 2017 7:23:37 PM
[info] Packaging /Users/kazuhirosera/github/oss/skinny-framework/common/target/scala-2.12/skinny-common_2.12-2.3.8-sources.jar ...
[info] Done packaging.
[info] Formatting 29 Scala sources {file:/Users/kazuhirosera/github/oss/skinny-framework/}common(compile) ...
[info] Wrote /Users/kazuhirosera/github/oss/skinny-framework/common/target/scala-2.12/skinny-common_2.12-2.3.8.pom
[info] :: delivering :: org.skinny-framework#skinny-common_2.12;2.3.8 :: 2.3.8 :: release :: Sat Jul 22 19:23:38 JST 2017
[info] Main Scala API documentation to /Users/kazuhirosera/github/oss/skinny-framework/common/target/scala-2.12/api...
[info] Compiling 29 Scala sources to /Users/kazuhirosera/github/oss/skinny-framework/common/target/scala-2.12/classes...
model contains 74 documentable templates
[warn] /Users/kazuhirosera/github/oss/skinny-framework/common/src/main/scala/skinny/nlp/SkinnyJapaneseAnalyzerFactory.scala:54: class CharArraySet in package util is deprecated: see corresponding Javadoc for more information.
[warn]       new CharArraySet(0, true),
[warn]           ^
[info] Main Scala API documentation successful.
[info] Packaging /Users/kazuhirosera/github/oss/skinny-framework/common/target/scala-2.12/skinny-common_2.12-2.3.8-javadoc.jar ...
[info] Done packaging.
[warn] one warning found
[info] Packaging /Users/kazuhirosera/github/oss/skinny-framework/common/target/scala-2.12/skinny-common_2.12-2.3.8.jar ...
[info] Done packaging.
Please enter PGP passphrase (or ENTER to abort): ************************
java.lang.IllegalStateException: Ivy file not found in cache for org.skinny-framework#skinny-common_2.12;2.3.8!
	at sbt.ResolutionCache.getResolvedModuleDescriptor(ResolutionCache.scala:52)
	at org.apache.ivy.core.deliver.DeliverEngine.deliver(DeliverEngine.java:111)
	at org.apache.ivy.Ivy.deliver(Ivy.java:601)
	at sbt.IvyActions$$anonfun$deliver$1.apply(IvyActions.scala:108)
	at sbt.IvyActions$$anonfun$deliver$1.apply(IvyActions.scala:103)
	at sbt.IvySbt$Module$$anonfun$withModule$1.apply(Ivy.scala:156)
	at sbt.IvySbt$Module$$anonfun$withModule$1.apply(Ivy.scala:156)
	at sbt.IvySbt$$anonfun$withIvy$1.apply(Ivy.scala:133)
	at sbt.IvySbt.sbt$IvySbt$$action$1(Ivy.scala:57)
	at sbt.IvySbt$$anon$4.call(Ivy.scala:65)
	at xsbt.boot.Locks$GlobalLock.withChannel$1(Locks.scala:93)
	at xsbt.boot.Locks$GlobalLock.xsbt$boot$Locks$GlobalLock$$withChannelRetries$1(Locks.scala:78)
	at xsbt.boot.Locks$GlobalLock$$anonfun$withFileLock$1.apply(Locks.scala:97)
	at xsbt.boot.Using$.withResource(Using.scala:10)
	at xsbt.boot.Using$.apply(Using.scala:9)
	at xsbt.boot.Locks$GlobalLock.ignoringDeadlockAvoided(Locks.scala:58)
	at xsbt.boot.Locks$GlobalLock.withLock(Locks.scala:48)
	at xsbt.boot.Locks$.apply0(Locks.scala:31)
	at xsbt.boot.Locks$.apply(Locks.scala:28)
	at sbt.IvySbt.withDefaultLogger(Ivy.scala:65)
	at sbt.IvySbt.withIvy(Ivy.scala:128)
	at sbt.IvySbt.withIvy(Ivy.scala:125)
	at sbt.IvySbt$Module.withModule(Ivy.scala:156)
	at sbt.IvyActions$.deliver(IvyActions.scala:103)
	at sbt.Classpaths$$anonfun$deliverTask$1.apply(Defaults.scala:1388)
	at sbt.Classpaths$$anonfun$deliverTask$1.apply(Defaults.scala:1388)
	at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:35)
	at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:34)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[error] (common/*:deliver) java.lang.IllegalStateException: Ivy file not found in cache for org.skinny-framework#skinny-common_2.12;2.3.8!
[error] Total time: 241 s, completed Jul 22, 2017 7:27:39 PM
```

**Comment by 3tty0n** [comment]
I also encountered the same problem.

This problem is happened in __1.0.0-RC8__ , but not happened in __1.0.0-RC7__ .

```
$ sbt +publishSigned
[info] Loading global plugins from /Users/izawa/.sbt/0.13/plugins/project
[info] Loading global plugins from /Users/izawa/.sbt/0.13/plugins
[info] Updating {file:/Users/izawa/.sbt/0.13/plugins/}global-plugins...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Loading project definition from /Users/izawa/.ghq/github.com/3tty0n/play-json-xml/project
[info] Set current project to play-json-xml (in build file:/Users/izawa/.ghq/github.com/3tty0n/play-json-xml/)
> +publishSigned
[info] Setting version to 2.12.2
[info] Reapplying settings...
[info] Set current project to play-json-xml (in build file:/Users/izawa/.ghq/github.com/3tty0n/play-json-xml/)
[info] Wrote /Users/izawa/.ghq/github.com/3tty0n/play-json-xml/target/scala-2.12/play-json-xml_2.12-0.2.1.pom
[info] :: delivering :: org.micchon#play-json-xml_2.12;0.2.1 :: 0.2.1 :: release :: Sun Jul 23 01:34:15 JST 2017
Please enter PGP passphrase (or ENTER to abort): ***********
[trace] Stack trace suppressed: run last *:deliver for the full output.
[error] (*:deliver) java.lang.IllegalStateException: Ivy file not found in cache for org.micchon#play-json-xml_2.12;0.2.1!
[error] Total time: 6 s, completed 2017/07/23 1:34:20
```

**Comment by alexarchambault** [comment]
I ran into this too.

As a workaround, doing a `publishLocal` before a `publishSigned` or `publishLocalSigned`, makes the latter two works.

I know [what changes](https://github.com/coursier/coursier/blob/d4cb232e65007871302b8a6c0c115ae89f8018b5/sbt-coursier/src/main/scala/coursier/CoursierPlugin.scala#L100-L101) caused this, that should be fixed in `1.0.0-RC9`.

**Comment by alexarchambault** [comment]
Should be fixed by https://github.com/coursier/coursier/pull/627, that will be included in `1.0.0-RC9`.

**Comment by alexarchambault** [comment]
1.0.0-RC9 was released a few days ago. Don"t hesitate to re-open if you still run into this.
